test/unittest/tests/block/test_block_file.cpp

```diff
@@ -0,0 +1,214 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [block_file]: block_open.c
+ * The block manager extent list consists of both extent blocks and size blocks. This unit test
+ * suite tests aims to test all of the allocation and frees of the extent and size block functions.
+ *
+ * The block session manages an internal caching mechanism for both block and size blocks that are
+ * created or discarded.
+ */
+#include "wt_internal.h"
+#include <catch2/catch.hpp>
+#include "../wrappers/mock_session.h"
+#include "../wrappers/config_parser.h"
+#include <iostream>
+
+const std::string ALLOCATION_SIZE = "512";
+const std::string BLOCK_ALLOCATION = "best";
+const std::string OS_CACHE_MAX = "0";
+const std::string OS_CACHE_DIRTY_MAX = "0";
+const std::string ACCESS_PATTERN = "random";
+const std::string DEFAULT_FILE_NAME = "test.txt";
+
+void
+free_block(WT_SESSION_IMPL *session, WT_BLOCK *block)
+{
+    WT_IGNORE_RET(__wti_bm_close_block(session, block));
+}
+
+void
+validate_block_fh(WT_BLOCK *block, uint expected_ref, std::string name)
+{
```

Comment:
The parameter `expected_ref` is not used in the function, so I suggest removing it.
The parameter `name` is passed by value which will (wastefully) create a unnecessary copy of the string when calling this function. `name` should be passed as a const reference to a string instead.